atc/resource/resource_test.go

```diff
@@ -1,50 +1,339 @@
-package resource_test
+package resource
```

Comment:
I think it was more because there wasn't an obvious benefit to using a `_test` package here. The [CodeReviewComments](https://github.com/golang/go/wiki/CodeReviewComments#import-dot) guidelines mentions, when discussing `import ( . package )`:

> The import . form can be useful in tests that, due to circular dependencies, cannot be made part of the package being tested

...which implies that `*_test` packages aren't just about testing the package via only its public interface - it's also (more?) about avoiding import cycles.

Looking at some large Go projects (including the Go standard library), from what I've seen, most of the time tests reside in the same package as the impl: e.g. [Nomad](https://github.com/hashicorp/nomad/blob/main/scheduler/annotate_test.go), [Kubernetes](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/daemon/daemon_controller_test.go), [Go standard library](https://github.com/golang/go/blob/master/src/net/addrselect_test.go).

That said, if you feel strongly, I don't mind changing it